src/main/java/Duke.java

```diff
@@ -5,50 +5,95 @@
-                        + tasks.get(taskNumber - 1).toString());
+                resolveTask(tasks, userInput);
             } else if ("todo".equals(userCommand)) {
-                String taskName = userInput.split(" ", 2)[1];
-                Todo newTodo = new Todo(taskName);
+                Todo newTodo = getTodoFromInput(userInput);
                 addTask(tasks, newTodo);
             } else if ("deadline".equals(userCommand)) {
-                String[] deadlineComponents = userInput.split("deadline | /by ");
-                String taskName = deadlineComponents[1];
-                String taskDeadline = deadlineComponents[2];
-                Deadline newDeadline = new Deadline(taskName, taskDeadline);
+                Deadline newDeadline = getDeadlineFromInput(userInput);
                 addTask(tasks, newDeadline);
             } else if ("event".equals(userCommand)) {
-                String[] eventComponents = userInput.split("event | /at ");
-                String eventName = eventComponents[1];
-                String eventDetails = eventComponents[2];
-                Event newEvent = new Event(eventName, eventDetails);
+                Event newEvent = getEventFromInput(userInput);
                 addTask(tasks, newEvent);
+            } else {
+                printRetryMessage();
             }
-            System.out.println("\t~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~");
+            printLineBorder();
             userInput = in.nextLine();
             userCommand = userInput.split(" ")[0];
         }
+    }
+
+    private static void printRetryMessage() {
+        System.out.println("Ohhh my sire please give me a valid command");
+    }
+
+    private static void printLineBorder() {
         System.out.println("\t~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~");
+    }
+
+    private static Event getEventFromInput(String userInput) {
+        String[] eventComponents = userInput.split("event | /at ");
+        String eventName = eventComponents[1];
+        String eventDetails = eventComponents[2];
+        Event newEvent = new Event(eventName, eventDetails);
+        return newEvent;
+    }
+
+    private static Deadline getDeadlineFromInput(String userInput) {
+        String[] deadlineComponents = userInput.split("deadline | /by ");
+        String taskName = deadlineComponents[1];
+        String taskDeadline = deadlineComponents[2];
+        Deadline newDeadline = new Deadline(taskName, taskDeadline);
+        return newDeadline;
+    }
+
+    private static Todo getTodoFromInput(String userInput) {
+        String taskName = userInput.split(" ", 2)[1];
+        Todo newTodo = new Todo(taskName);
+        return newTodo;
+    }
+
+    private static void resolveTask(ArrayList<Task> tasks, String userInput) {
+        int taskNumber = Integer.parseInt(userInput.split(" ")[1]);
+        tasks.get(taskNumber - 1).markAsDone();
```

Comment:
Correct me if I am wrong, but I think the spacing on both ends of the sign is not necessary. Otherwise I will suggest it to be standardized with the rest of the mathematical expressions.

Reply:
Thanks for your kind review! May i ask if you're referring to the - sign?

Reply:
Yes Randy. Its is a minor formatting issue that I have found, if it is at all an issue in the first place.